src/adapters/ws/client.ts

```diff
@@ -77,13 +77,12 @@ class WsClientAdapter extends Adapter {
   private getWsChannels() {
     const channels = []
     for (const channel of this.channelNames) {
-      if (this.parsedAsyncAPI.channel(channel).hasBinding('ws')) {
-        if (this.parsedAsyncAPI.channel(channel).hasServers()) {
+      if (this.parsedAsyncAPI.channels().get(channel).bindings().get('ws')) {
+        if (this.parsedAsyncAPI.channels().get(channel).servers().length !== 0) { // NOSONAR
```

Comment:
Is this working for you? 🤔 

I think it should be:

```suggestion
        if (this.parsedAsyncAPI.channels().get(channel).servers().all().length !== 0) { // NOSONAR
```